lib/utils/network_utils.dart

```diff
@@ -0,0 +1,133 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/l10n.dart';
+import 'package:matrix/matrix.dart';
+import 'package:tawkie/config/themes.dart';
+
+import 'matrix_sdk_extensions/matrix_locals.dart';
+
+bool containsFacebook(List<String> participantsIds) {
+  return participantsIds.any((id) => id.contains('@messenger2'));
+}
+
+bool containsInstagram(List<String> participantsIds) {
+  return participantsIds.any((id) => id.contains('@instagram2_'));
+}
+
+bool containsWhatsApp(List<String> participantsIds) {
+  return participantsIds.any((id) => id.contains('@whatsapp'));
+}
+
+bool containsLinkedin(List<String> participantsIds) {
+  return participantsIds.any((id) => id.contains('@linkedin'));
+}
+
+bool containsDiscord(List<String> participantsIds) {
+  return participantsIds.any((id) => id.contains('@discord'));
+}
+
+bool containsSignal(List<String> participantsIds) {
+  return participantsIds.any((id) => id.contains('@signal'));
+}
+
+String removeFacebookTag(String displayname) {
+  if (displayname.contains('(FB)')) {
+    displayname = displayname.replaceAll('(FB)', ''); // Delete (FB)
+  }
+  return displayname;
+}
+
+String removeInstagramTag(String displayname) {
+  if (displayname.contains('(IG)')) {
+    displayname = displayname.replaceAll('(IG)', ''); // Delete (Instagram)
+  }
+  return displayname;
+}
+
+String removeWhatsAppTag(String displayname) {
+  if (displayname.contains('(WA)')) {
+    displayname = displayname.replaceAll('(WA)', ''); // Delete (WA)
+  }
+  return displayname;
+}
+
+String removeLinkedinTag(String displayname) {
+  if (displayname.contains('(LinkedIn)')) {
+    displayname = displayname.replaceAll('(LinkedIn)', ''); // Delete (LinkedIn)
+  }
+  return displayname;
+}
+
+String removeDiscordTag(String displayname) {
+  if (displayname.contains('(Discord)')) {
+    displayname = displayname.replaceAll('(Discord)', ''); // Delete (Discord)
+  }
+  return displayname;
+}
+
+String removeSignalTag(String displayname) {
+  if (displayname.contains('(Signal)')) {
+    displayname = displayname.replaceAll('(Signal)', ''); // Delete (Signal)
+  }
+  return displayname;
+}
+
+Future<List<dynamic>> loadRoomInfo(BuildContext context, Room room) async {
+  List<User> participants = room.getParticipants();
+  Color? networkColor;
+  Image? networkImage;
+  final participantsIds = participants.map((member) => member.id).toList();
```

Comment:
This leads to poor performance and increases page load time. It makes a copy of the list every time, and we don't need the entire list every time. Please refactor the `containsNetwork` functions to accept the `List<User>` as parameters.